R/produce_slf_homelessness_tests.R

```diff
@@ -0,0 +1,43 @@
+#' SLF Homelessness Extract Tests
+#'
+#' @param data The data for testing
+#' @param max_min_vars Shouldn't need to change, currently specifies keydate1 and deydate2
+#'
+#' @description Produce the tests for the SLF Homelessness Extract
+#'
+#' @return
+#'
+#' @export
+#' @family slf test functions
+produce_slf_homelessness_tests <- function(data,
+                                           max_min_vars = c("record_keydate1", "record_keydate2")) {
+  test_flags <- data %>%
+    # Would usually use create_demog_test_flags() here but needs to be different for Homelessness
+    dplyr::arrange(.data$chi) %>%
+    # create test flags
+    dplyr::mutate(
+      has_chi = dplyr::if_else(!is_missing(.data$chi), 1, 0),
+      n_males = dplyr::if_else(.data$gender == 1, 1, 0),
+      n_females = dplyr::if_else(.data$gender == 2, 1, 0),
+      missing_dob = dplyr::if_else(is.na(.data$dob), 1, 0),
```

Comment:
Suggestion to use `create_demog_test_flags` here. We spoke about this at the R conversion meeting. Just highlighting as a reminder when you have the time to go back to this :)

Reply:
Used `create_demog_test_flags` here instead